tests/install/test_distribution.pm

```diff
@@ -9,7 +9,24 @@ sub run {
     diag('initialize working copy of openSUSE tests distribution with correct user');
     assert_script_run('username=bernhard email=bernhard@susetest /usr/share/openqa/script/fetchneedles', 3600);
     save_screenshot;
-    assert_script_run('zypper --no-refresh -n in os-autoinst-distri-opensuse-deps', 600);
+
+    # 104 is the zypper exit code if a package cannot be found
+    my $cmd = <<'EOM';
+(
+    for i in {1..7}
+    do
+        zypper --non-interactive install os-autoinst-distri-opensuse-deps && exit 0
+        rc=$?
+        (( $rc == 104 )) || break
+    done
+    exit $rc
```

Comment:
Is this actually needed? If you just "break" from the loop the exit code should be preserved. The assignment `rc=$?` would break that. So would the following work?:

```sh
for i in {1..7}; do
    zypper --non-interactive install os-autoinst-distri-opensuse-deps && break || ( [ $? = 104 ] || break)
done
```

Reply:
`( [ $? = 104 ] || break)` does not wiork because the `break` is in the subshell